test/distributed/fsdp/test_fsdp_use_orig_params.py

```diff
@@ -0,0 +1,875 @@
+        ):
+            fsdp_model = fsdp_model.cuda()
+        return fsdp_model, fsdp_optim
+
+    def _check_train_parity(
+        self,
+        ddp_model: DDP,
+        ddp_optim: torch.optim.Optimizer,
+        fsdp_model: FSDP,
+        fsdp_optim: torch.optim.Optimizer,
+        set_to_none: bool,
+        num_iters: int = 10,
+    ):
+        """Checks training parity between DDP and FSDP."""
+        device = torch.device("cuda")
+        for i in range(num_iters):
+            iter_losses = []
+            for model, optim in ((ddp_model, ddp_optim), (fsdp_model, fsdp_optim)):
+                module = model.module
+                # Test two different `zero_grad()` timings
+                if i % 2 == 0:
+                    optim.zero_grad(set_to_none=set_to_none)  # pre-forward
+                inp = module.get_input(device)
+                output = model(*inp)
+                loss = module.get_loss(inp, output).to(device)
+                iter_losses.append(loss)
+                if i % 2 == 1:
+                    optim.zero_grad(set_to_none=set_to_none)  # pre-backward
+                module.run_backward(loss)
+                # Perform the DDP optimizer step on CPU to match FSDP if needed
+                if model is ddp_model and fsdp_model.cpu_offload.offload_params:
+                    model.to(torch.device("cpu"))
+                optim.step()
+                if model is ddp_model and fsdp_model.cpu_offload.offload_params:
+                    model.to(torch.device("cuda"))
+            torch.testing.assert_close(iter_losses[0], iter_losses[1])
+            iter_losses.clear()
+        atol = 1e-5 if self.world_size <= 2 else 1e-4
+        rtol = 1e-6 if self.world_size <= 2 else 1e-5
+        with FSDP.summon_full_params(fsdp_model):
+            for p1, p2 in zip(ddp_model.parameters(), fsdp_model.parameters()):
+                torch.testing.assert_close(p1, p2, atol=atol, rtol=rtol)
+
+    def _get_sharding_strategy_from_str(
+        self, sharding_strategy_str: str
+    ) -> ShardingStrategy:
+        if sharding_strategy_str == "no_shard":
+            sharding_strategy = ShardingStrategy.NO_SHARD
+        elif sharding_strategy_str == "shard_grad_op":
+            sharding_strategy = ShardingStrategy.SHARD_GRAD_OP
+        elif sharding_strategy_str == "full_shard":
+            sharding_strategy = ShardingStrategy.FULL_SHARD
+        else:
+            raise ValueError(f"Invalid string: {sharding_strategy_str}")
+        return sharding_strategy
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize(
+        "sharding_strategy_str",
+        ["no_shard", "shard_grad_op", "full_shard"],
+    )
+    def test_diff_hyperparams(self, sharding_strategy_str: str):
+        """
+        Tests FSDP parity with DDP when using multiple parameter groups with
+        different hyperparameter settings.
+        """
+        sharding_strategy = self._get_sharding_strategy_from_str(sharding_strategy_str)
```

Comment:
why not just parametrize based on the ShardingStrategy enum and remove the getter from string? also, use self.subTest here for the sharding strategy as well?